### PR TITLE
feat

### PR DESCRIPTION
**Pull request by piotrSobie**

### CODE REVIEW

File: src/app/header/header.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<nav class="navbar navbar-default">
+  <div class="container-fluid">
+    <div class="navbar-header">
+      <a routerLink="/" class="navbar-brand">Toggl</a>
+    </div>
+    <div class="collapse navbar-collapse">
+      <ul class="nav navbar-nav navbar-right">
```
Comment (WrobelAdrian) [inline src/app/header/header.component.html]: Links should be generated dynamically (use *ngFor)

File: src/app/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      <div class="form-group">
+        <label for="password">PASSWORD</label>
+        <input
+         type="password"
+         id="password"
+         class="form-control"
+         ngModel
+         name="password"
+         required
+         minlength="6"
```
Comment (WrobelAdrian) [inline src/app/login/login.component.html]: Add custom password validator https://angular.io/guide/form-validation and notifications on password and email inputs (so user will be notified if he typed wrong password or email)

File: src/app/register/register.component.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <div class="form-group">
+        <label for="password">SET PASSWORD</label>
+        <input
+          type="password"
+          id="password"
+          class="form-control"
+          ngModel
+          name="password"
+          required
+          minlength="6"
```
Comment (WrobelAdrian) [inline src/app/register/register.component.html]: Add custom password validator https://angular.io/guide/form-validation and notifications on password and email inputs (so user will be notified if he typed wrong password or email)

File: src/app/header/header.component.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<nav class="navbar navbar-default">
```
Comment (WrobelAdrian) [inline src/app/header/header.component.html]: Create shared module https://angular.io/guide/sharing-ngmodules (find more complex example via google). Move header to shared module + create footer component (also in shared module)

File: src/app/app.module.ts [hunk 5]
```diff
@@ -2,13 +2,28 @@ import { BrowserModule } from '@angular/platform-browser';
...
+    LoginComponent,
+    RegisterComponent,
+    TrainingComponent,
+    PricingComponent,
+    FeaturesComponent
   ],
   imports: [
-    BrowserModule
+    BrowserModule,
+    FormsModule,
```
Comment (WrobelAdrian) [inline src/app/app.module.ts]: Browser, and Forms module move to shared module

File: src/app/features/features.component.html [hunk 6]
```diff
@@ -0,0 +1 @@
+<p>features works!</p>
```
Comment (WrobelAdrian) [inline src/app/features/features.component.html:1]: features component is not part of loginAndRegister feature. Should be commited to different branch.

File: src/app/header/header.component.html (hunk 4 above)
Comment (WrobelAdrian) [inline src/app/header/header.component.html]: header component is not part of loginAndRegister feature. Should be commited to different branch.

File: src/app/pricing/pricing.component.html [hunk 7]
```diff
@@ -0,0 +1 @@
+<p>pricing works!</p>
```
Comment (WrobelAdrian) [inline src/app/pricing/pricing.component.html:1]: pricing component is not part of loginAndRegister feature. Should be commited to different branch.

File: src/app/training/training.component.html [hunk 8]
```diff
@@ -0,0 +1 @@
+<p>training works!</p>
```
Comment (WrobelAdrian) [inline src/app/training/training.component.html:1]: training component is not part of loginAndRegister feature. Should be commited to different branch.

File: src/app/header/header.component.html (hunk 1 above)
Comment (piotrSobie) [reply src/app/header/header.component.html]: done

File: src/app/login/login.component.html (hunk 2 above)
Comment (piotrSobie) [reply src/app/login/login.component.html]: done

File: src/app/register/register.component.html (hunk 3 above)
Comment (piotrSobie) [reply src/app/register/register.component.html]: done

File: src/app/header/header.component.html (hunk 4 above)
Comment (piotrSobie) [reply src/app/header/header.component.html]: done

File: src/app/app.module.ts (hunk 5 above)
Comment (piotrSobie) [reply src/app/app.module.ts]: done

File: src/app/shared/shared.module.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+import { CommonModule } from '@angular/common';
+import {BrowserModule} from "@angular/platform-browser";
+import {FormsModule} from "@angular/forms";
+
+
+
+@NgModule({
+  declarations: [],
+  imports: [],
+  exports: [
```
Comment (WrobelAdrian) [inline src/app/shared/shared.module.ts:19]: To export modules, you need to import them as well

File: src/app/app.module.ts [hunk 10]
```diff
@@ -1,14 +1,31 @@
...
-    AppComponent
+    AppComponent,
+    HeaderComponent,
+    LoginComponent,
+    RegisterComponent,
+    TrainingComponent,
+    PricingComponent,
+    FeaturesComponent,
+    FooterComponent,
+    PasswordValidatorDirective
```
Comment (WrobelAdrian) [inline src/app/app.module.ts:20]: Header and footer components should be imported / exported via shared module

**Pull request merged**